js/stanford_paragraph_types.p_hero.js

```diff
@@ -3,44 +3,131 @@
  * Applies some functionality on the paragraph types.
  */
 
+/**
+ * Called automatically by YouTube's API.
+ */
+function onYouTubeIframeAPIReady() {
```

Comment:
Perhaps pedantic, but shouldn't this be placed after the definition of the function it is calling? I know this is called on an event but it may save you some headache later if this code gets refactored.

Reply:
then would YT's api be able to call the function if its in the scope of drupal behaviors?

Reply:
I'm just talking about placing this function after ` Drupal.behaviors.stanfordParagraphPHero.youTubeReady` as that function call is in the body of this event callback.

Reply:
So you mean put this at the bottom of the file? it has to be at the first level of the file or it doesn't get called by YT.  even just putting it inside `(function ($) {` prevents it from being called.

Reply:
Yep, don't encapsulate it in an anon function. You have to keep it global, I was suggesting moving it to the very bottom of the file outside of the `(function ($)` wrapper.

Reply:
It is the difference of 
```

function b() {
 return a();
}

function a() {
 return 1;
}
```

and

``` 
function a() {
  return 1;
}

function b() {
  return a();
}
```

Where the second one is the desired one.